kdbg/prefmisc.cpp

```diff
@@ -25,9 +25,14 @@ PrefMisc::PrefMisc(QWidget* parent) :
     m_grid.addWidget(&m_popForeground, 0, 0, 1, 2);
     m_grid.addItem(new QSpacerItem(0, m_popForeground.sizeHint().height()), 0, 0);
 
-    m_backTimeoutLabel.setText(i18n("Time until window goes &back (in milliseconds):"));
+    m_backTimeoutLabel.setText(i18n("Time until window goes &back [ms] (0 = off):"));
+    const QString backTimeoutTT_Cqs( i18n("Time in ms. 0 has the special "
+	        "meaning of not actively going into the background, but raise it in "
+		    "case of a stop of the debuge.") );
+    m_backTimeoutLabel.setToolTip( backTimeoutTT_Cqs );
     m_backTimeoutLabel.setMinimumSize(m_backTimeoutLabel.sizeHint());
```

Comment:
If we can have a special user-visible value "never", then we do not need an explanation. The special value can just be a very large value, say 1'000'000ms, then we do not even need special code for the timer, it just takes 16 minutes until the debugger goes into the background. That should be sufficient for "never".

If you do want to add an explanation, you can drop the bit about "raise it", because that happens also when the timeout is not the special value.

Reply:
I like that window is raised if the program stops. To have a timer running for feature  the user didn't like have, I often like to see at least a part of the kdbg window, to can activate it through a mouse click instead of search it at the taskbar, because some other windows which is much more seldom used are in front of it. Instead of 0, it would also be possible to take a negative value, which is seeing it as unsigned int is a big value. The idea behind 0 was also the short description what is possible through that value,.
Do you really prefer a ComboBox LineEdit combination for that?  The user expectation could be, that he likes see the previous used values too.

Reply:
I think that raise-to-top happens always even when there is a timeout. (At least that is the intent and I think it worked when I implemented it some decades ago.) Therefore, in this regard your new setting is not special. It's only special because it does not send KDbg into the background.

Whether it is a Combobox or something else I do not care much. I only saw some controls where it was possible to specify numbers and in addition one special, named value. I also would not mind to make -1 the special value for your new behavior, but in the user-interface it should be this one named special value that I propose to be "never".

Reply:
I thought about it. What do you think if we replace the QLabel with a QCheckbox that it looks like this:
  [x] lower window after [ms]: ___200_
If the checkbox is unchecked m_backTimeout is set to the negative value of checked m_backTimeout) minus 1. 
In this way no addition configuration value would be needed.

Reply:
The checkbox is fine, too.

Concerning the configuration value, we have to be careful and keep in mind that the setting can be read by old versions of KDbg. Will they do something useful with a negative timeout, or will they be confused? I think that a new configuration value for the checkbox is better.